src/server/shell/basic_accessibility_manager.cpp

```diff
@@ -102,39 +64,70 @@ void mir::shell::BasicAccessibilityManager::repeat_rate_and_delay(
 
 void mir::shell::BasicAccessibilityManager::mousekeys_enabled(bool on)
 {
-    if (on)
-        transformer.add_registration();
-    else
-        transformer.remove_registration();
+    if (on && !mouse_keys_transformer->is_enabled())
+    {
+        event_transformer->append(mouse_keys_transformer);
+        mouse_keys_transformer->enable();
+    }
```

Comment:
This is racy: Two threads can call `is_enabled()` before either calls `enable()` resulting in multiple calls to `append()`

The previous `compare_exchange_strong()` implementation avoided this